doc/news/OSW-725.misc.rst

```diff
@@ -0,0 +1 @@
+Added mypy type hints.
```

Comment:
And ruff and towncrier. Also mention here that you disabled black, flake8 and isort if you decide to do so.